### PR TITLE
Only handle refunds when there are payments, otherwise skip

### LINKED ISSUE #6969: [CAL-1025] Bookings can't be rejected
Try to make a booking that requires confirmation and reject it from the bookings tab

Created via Threads. See full discussion: [https://threads.com/34447694005](https://threads.com/34447694005)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1025](https://linear.app/calcom/issue/CAL-1025/bookings-cant-be-rejected)</sub>

### PR DESCRIPTION
**Pull request by emrysal**
## What does this PR do?

Fixes #6969

### CODE REVIEW

File: apps/web/components/booking/BookingListItem.tsx [hunk 1]
```diff
@@ -246,23 +246,26 @@ function BookingListItem(booking: BookingItemProps) {
...
-            {t("rejection_reason")}
-            <span className="font-normal text-gray-500"> (Optional)</span>
-          </p>
-          <TextArea
-            name={t("rejection_reason")}
-            value={rejectionReason}
-            onChange={(e) => setRejectionReason(e.target.value)}
-            className="mb-5 sm:mb-6"
-          />
+          <div>
```
Comment (emrysal) [inline apps/web/components/booking/BookingListItem.tsx:249]: Just fixing some layout issues

File: packages/trpc/server/routers/viewer/bookings.tsx [hunk 2]
```diff
@@ -1077,69 +1079,73 @@ export const bookingsRouter = router({
           },
         });
       } else {
-        const successPayment = booking.payment.find((payment) => payment.success);
-        if (!successPayment) {
-          throw new Error("Cannot reject a booking without a successful payment");
-        }
+        // handle refunds
+        if (!!booking.payment.length) {
```
Comment (emrysal) [inline packages/trpc/server/routers/viewer/bookings.tsx:1083]: Only execute refund logic when there is payments on a booking.

File: apps/web/components/booking/BookingListItem.tsx [hunk 3]
```diff
@@ -246,23 +246,26 @@ function BookingListItem(booking: BookingItemProps) {
       {/* NOTE: Should refactor this dialog component as is being rendered multiple times */}
       <Dialog open={rejectionDialogIsOpen} onOpenChange={setRejectionDialogIsOpen}>
         <DialogContent>
```
Comment (sean-brydon) [inline apps/web/components/booking/BookingListItem.tsx:248]: NIT Dialog content actually accepts props to get this looking as per spec
CleanShot 2023-02-09 at 15 21 15@2x

File: packages/trpc/server/routers/viewer/bookings.tsx [hunk 4]
```diff
@@ -774,6 +774,8 @@ export const bookingsRouter = router({
 
     /** When a booking that requires payment its being confirmed but doesn't have any payment,
      * we shouldn’t save it on DestinationCalendars
+     *
+     * FIXME: This can cause unintended confirmations on rejection.
```
[the remark this answers is not in the document]
Comment (zomars) [reply packages/trpc/server/routers/viewer/bookings.tsx:778]: Cc @alannnc

**Review by zomars — APPROVED** [review]

**Pull request merged**